### PR TITLE
Replace flake8 with ruff & cython-lint

### PR DESCRIPTION
**Pull request by charlesbluca**
This brings our Python/Cython linting somewhat more in-line with that of cuDF, as well as giving us some additional Cython checks that weren't being done with flake8.

### CODE REVIEW

**Review by pentschev — APPROVED** [review]
This looks great @charlesbluca , thanks for it. I left one question out of curiosity only, otherwise LGTM.

File: python/ucxx/_lib/libucxx.pyx [hunk 1]
```diff
@@ -302,6 +299,9 @@ cdef class UCXContext():
         cdef ConfigMap cpp_config_in, cpp_config_out
         cdef dict context_config
 
+        if config_dict is None:
+            config_dict = {}
+
```
[the remark this answers is not in the document]
Comment (charlesbluca) [reply python/ucxx/_lib/libucxx.pyx:304]: This was done manually - does this seem like a suitable alternative to the mutable default arg?

[the remark this answers is not in the document]
Comment (pentschev) [reply python/ucxx/_lib/libucxx.pyx:304]: Sure, that's in line with what we do in [other places](https://github.com/rapidsai/ucxx/blob/branch-0.36/python/ucxx/_lib/libucxx.pyx#L649-L650). I was primarily curious if the linter was _that smart_ or you actually intervened manually. 🙂

**Pull request merged**